tests/chainerx_tests/unit_tests/routines_tests/test_logic.py

```diff
@@ -205,3 +205,105 @@ def forward_xp(self, inputs, xp):
         a, = inputs
         b = xp.logical_not(a)
         return b,
+
+
+# Hack for Issue #6778
+# TO-DO(kshitij12345): Remove when issue fixed
+def cast_if_float16(xp, a):
+    if a.dtype == 'float16':
+        return xp.logical_not(a.astype(bool, True))
+    else:
+        return a.astype(bool, True)
+
+
+def logical_and_lambda(xp, a, b):
+    return xp.logical_and(a, b)
+
+
+def logical_or_lambda(xp, a, b):
+    return xp.logical_and(a, b)
+
+
+_binary_logical_params = \
+    chainer.testing.product({
+        'dtypes': _expected_all_dtypes_comparison,
+        'func': [
+            logical_and_lambda, logical_or_lambda
+        ],
+        'inputs': [
+            ([], []),
+            ([True], [True]),
+            ([True], [False]),
+        ]
+    }) + chainer.testing.product({
+        'dtypes': _expected_numeric_dtypes_comparison,
+        'func': [
+            logical_and_lambda, logical_or_lambda
+        ],
+        'inputs': [
+            ([0], [0]),
+            ([0], [-0]),
+            ([0], [1]),
+            ([0, 1, 2], [0, 1, 2]),
+            ([1, 1, 2], [0, 1, 2]),
+            ([0, 1, 2], [1, 2, 3]),
+            ([[0, 1], [2, 3]], [[0, 1], [2, 3]]),
+            ([[0, 1], [2, 3]], [[0, 1], [2, -2]]),
+            ([[0, 1], [2, 3]], [[1, 2], [3, 4]]),
+            (0, [0]),
+            (1, [0]),
+            ([], [0]),
+            ([0], [[0, 1, 2], [3, 4, 5]]),
+            ([[0], [1]], [0, 1, 2]),
+            ([0.2], [0.2]),
+            ([0.2], [-0.3]),
+        ],
+    }) + chainer.testing.product({
+        'dtypes': _expected_float_dtypes_comparison,
+        'func': [
+            logical_and_lambda, logical_or_lambda
+        ],
+        'inputs': [
+            ([0., numpy.nan], [0., 1.]),
+            ([0., numpy.nan], [0., numpy.nan]),
+            ([0., numpy.inf], [0., 1.]),
+            ([0., -numpy.inf], [0., 1.]),
+            ([numpy.inf, 1.], [numpy.inf, 1.]),
+            ([-numpy.inf, 1.], [-numpy.inf, 1.]),
+            ([numpy.inf, 1.], [-numpy.inf, 1.]),
+            ([numpy.inf, 1.], [-numpy.inf, numpy.nan]),
+        ]
+    })
+
+
+@op_utils.op_test(['native:0', 'cuda:0'])
+@chainer.testing.parameterize(*(
+    _binary_logical_params
+))
+# Ignore warnings from numpy for NaN comparisons.
+@pytest.mark.filterwarnings('ignore:invalid value encountered in ')
+class TestLogicalBinary(op_utils.NumpyOpTest):
+
+    skip_backward_test = True
+    skip_double_backward_test = True
+
+    def generate_inputs(self):
+        a_object, b_object = self.inputs
+        a_dtype, b_dtype = self.dtypes
+        a = numpy.array(a_object, a_dtype)
+        b = numpy.array(b_object, b_dtype)
+        return a, b
+
+    def forward_xp(self, inputs, xp):
+        a, b = inputs
+
+        def on_cuda(a):
+            return 'cuda' in str(a.device)
```

Comment:
You can get the backend name "cuda" via `a.device.backend.name`.

Reply:
Since #6780  was merged, will remove this

Reply:
Btw, thanks for the suggestion. Will keep that in mind for future.